server/src/test/java/org/spine3/server/stand/StandShould.java

```diff
@@ -444,6 +452,199 @@ private static ProjectId projectIdFor(int numericId) {
+                super.onNext(value);
+
+                final List<Any> messages = value.getMessagesList();
+                assertFalse(messages.isEmpty());
+
+                final Customer sampleCustomer = getSampleCustomer();
+                final Customer customer = AnyPacker.unpack(messages.get(0));
+                assertTrue(customer.getName()
+                                   .equals(sampleCustomer.getName()));
+                assertFalse(customer.hasId());
+                assertTrue(customer.getNicknamesList().isEmpty());
+            }
+        });
+    }
+
+    @Test
+    public void retrieve_collection_fields_if_required() {
+        requestSampleCustomer(new int[] {Customer.NICKNAMES_FIELD_NUMBER - 1}, new MemoizeQueryResponseObserver() {
+            @Override
+            public void onNext(QueryResponse value) {
+                super.onNext(value);
+
+                final List<Any> messages = value.getMessagesList();
+                assertFalse(messages.isEmpty());
+
+                final Customer sampleCustomer = getSampleCustomer();
+                final Customer customer = AnyPacker.unpack(messages.get(0));
+                assertEquals(customer.getNicknamesList(), sampleCustomer.getNicknamesList());
+
+                assertFalse(customer.hasName());
+                assertFalse(customer.hasId());
+            }
+        });
+    }
+
+    @Test
+    public void retrieve_all_requested_fields() {
+        requestSampleCustomer(new int[] {Customer.NICKNAMES_FIELD_NUMBER - 1, Customer.ID_FIELD_NUMBER - 1}, new MemoizeQueryResponseObserver() {
+            @Override
+            public void onNext(QueryResponse value) {
+                super.onNext(value);
+
+                final List<Any> messages = value.getMessagesList();
+                assertFalse(messages.isEmpty());
+
+                final Customer sampleCustomer = getSampleCustomer();
+                final Customer customer = AnyPacker.unpack(messages.get(0));
+                assertEquals(customer.getNicknamesList(), sampleCustomer.getNicknamesList());
+
+                assertFalse(customer.hasName());
+                assertTrue(customer.hasId());
+            }
+        });
+    }
+
+    @Test
+    public void retrieve_whole_entity_if_nothing_is_requested() {
+        //noinspection ZeroLengthArrayAllocation
+        requestSampleCustomer(new int[] {}, getDuplicateCostumerStreamObserver());
+    }
+
+    @Test
+    public void handle_mistakes_in_query_silently() {
+        //noinspection ZeroLengthArrayAllocation
+        final Stand stand = prepareStandWithAggregateRepo(InMemoryStandStorage.newBuilder().build());
+
+        final Customer sampleCustomer = getSampleCustomer();
+
+        stand.update(sampleCustomer.getId(), AnyPacker.pack(sampleCustomer));
+
+        // FieldMask with invalid type URLs.
+        final String[] paths = {"blablabla", Project.getDescriptor().getFields().get(2).getFullName()};
```

Comment:
Please do not use `"blablabla"` as a string literal. Instead, name it after the thing you reference.

Use `"invalid_type_url_example"` in this case.